test/resources/orderCreation.ts

```diff
@@ -0,0 +1,26 @@
+import { Wallet, Contract, BigNumber } from "ethers";
+
+import { Order } from "../../src/js/orders.spec";
+
+export const orderAlwaysValid = function (
+  sellAmount: BigNumber | number,
+  buyAmount: BigNumber | number,
+  sellToken: Contract,
+  buyToken: Contract,
+  wallet: Wallet,
+  nonce: BigNumber | number,
+): Order {
+  const beginningOfTime = 0;
+  const endOfTime = 2 ** 32 - 1;
```

Comment:
From @josojo's comment above, should we use a sentinel value of 0 as "no validity", this would also make setting never expiring orders cheaper?

Reply:
Difference in gas cost per order in the test `clears baseTestInput - two generic overlapping orders` with an extra check for the zero sentinel.
| Gas cost per order current master | Diff, order with zero validity | Diff, order with validity set |
| --------------- | --------------- | --------------- |
| 152778 | -81 | 49 |

So for this change to be meaningful we assume that at least about 40% of the orders won't be specifying any validity.

Reply:
Then I don't think its worth it. I expect it to be set quite often.